src/drive/web/modules/views/Folder/FolderViewBody.jsx

```diff
@@ -152,25 +155,41 @@ const FolderViewBody = ({
           {hasDataToShow &&
             !needsToWait && (
               <div className={!isDesktop ? 'u-ov-hidden' : ''}>
-                {queryResults.map((query, queryIndex) => (
-                  <React.Fragment key={queryIndex}>
-                    {query.data.map(file => (
-                      <File
-                        key={file._id}
-                        attributes={file}
-                        withSelectionCheckbox
-                        onFolderOpen={navigateToFolder}
-                        onFileOpen={handleFileOpen}
-                        withFilePath={withFilePath}
-                        thumbnailSizeBig={isBigThumbnail}
-                        actions={actions}
-                        refreshFolderContent={refreshFolderContent}
-                        isInSyncFromSharing={isInSyncFromSharing()}
-                      />
-                    ))}
-                    {query.hasMore && <LoadMore fetchMore={query.fetchMore} />}
-                  </React.Fragment>
-                ))}
+                <>
+                  {syncingFakeFile && (
+                    <File
+                      attributes={syncingFakeFile}
+                      withSelectionCheckbox={false}
+                      onFolderOpen={() => {}}
+                      onFileOpen={() => {}}
+                      actions={[]}
+                      isInSyncFromSharing={true}
+                    />
+                  )}
+                  {queryResults.map((query, queryIndex) => (
+                    <React.Fragment key={queryIndex}>
+                      {query.data.map(file => (
+                        <File
+                          key={file._id}
+                          attributes={file}
+                          withSelectionCheckbox
+                          onFolderOpen={navigateToFolder}
+                          onFileOpen={handleFileOpen}
+                          withFilePath={withFilePath}
+                          thumbnailSizeBig={isBigThumbnail}
+                          actions={actions}
+                          refreshFolderContent={refreshFolderContent}
+                          isInSyncFromSharing={
+                            !isSharingContextEmpty && isSharingShorcut(file)
```

Comment:
we should probably add an usecallback to isSharingShortcut. Having it executed everytime on each file when the component is re-render makes me uncomfortable

Reply:
Hmmm en fait je vois pas bien... `file` n'est pas en lien direct avec le re-rendu du composant, et n'est donc pas une dépendance correct pour un useCallback. Ce serait plutôt éventuellement `queryResults` à utiliser en dépendance, mais ici on est sur un file de `queryResults` du coup je vois pas trop quelle fonction on pourrait memoiser... ?

Reply:
Pardon j'ai dit useCallback mais clairement je pensais à mémoiser le retour de isSharingShortcut(file), histoire de ne pas avoir à checker le content de l'objet file à chaque fois et de taper le "cache".

Reply:
le lodash memoize crée une erreur webpack à cet endroit, je vais faire sans.